flink-contrib/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/AbstractRocksDBState.java

```diff
@@ -132,55 +132,91 @@ public void setCurrentNamespace(N namespace) {
+			backend.getCurrentKey(),
+			currentNamespace,
+			this.keySerializationStream,
+			this.keySerializationDateDataOutputView);
 	}
 
-	protected void writeKeyWithGroupAndNamespace(int keyGroup, K key, N namespace) throws IOException {
+	protected void writeKeyWithGroupAndNamespace(
+			int keyGroup, K key, N namespace,
+			final ByteArrayOutputStreamWithPos keySerializationStream,
+			final DataOutputView keySerializationDateDataOutputView) throws IOException {
+
 		keySerializationStream.reset();
-		writeKeyGroup(keyGroup);
-		writeKey(key);
-		writeNameSpace(namespace);
+		writeKeyGroup(keyGroup, keySerializationDateDataOutputView);
+		writeKey(key, keySerializationStream, keySerializationDateDataOutputView);
+		writeNameSpace(namespace, keySerializationStream, keySerializationDateDataOutputView);
 	}
 
-	private void writeKeyGroup(int keyGroup) throws IOException {
+	private void writeKeyGroup(
+			int keyGroup, final DataOutputView keySerializationDateDataOutputView)
+			throws IOException {
+
 		for (int i = backend.getKeyGroupPrefixBytes(); --i >= 0;) {
 			keySerializationDateDataOutputView.writeByte(keyGroup >>> (i << 3));
 		}
 	}
 
-	private void writeKey(K key) throws IOException {
+	private void writeKey(
+			K key, final ByteArrayOutputStreamWithPos keySerializationStream,
+			final DataOutputView keySerializationDateDataOutputView) throws IOException {
+
 		//write key
 		int beforeWrite = keySerializationStream.getPosition();
 		backend.getKeySerializer().serialize(key, keySerializationDateDataOutputView);
 
 		if (ambiguousKeyPossible) {
 			//write size of key
-			writeLengthFrom(beforeWrite);
+			writeLengthFrom(beforeWrite, keySerializationStream,
+				keySerializationDateDataOutputView);
 		}
 	}
 
-	private void writeNameSpace(N namespace) throws IOException {
+	private void writeNameSpace(
+			N namespace, final ByteArrayOutputStreamWithPos keySerializationStream,
+			final DataOutputView keySerializationDateDataOutputView) throws IOException {
+
 		int beforeWrite = keySerializationStream.getPosition();
 		namespaceSerializer.serialize(namespace, keySerializationDateDataOutputView);
 
 		if (ambiguousKeyPossible) {
 			//write length of namespace
-			writeLengthFrom(beforeWrite);
+			writeLengthFrom(beforeWrite, keySerializationStream,
+				keySerializationDateDataOutputView);
 		}
 	}
 
-	private void writeLengthFrom(int fromPosition) throws IOException {
+	private static void writeLengthFrom(
+			int fromPosition, final ByteArrayOutputStreamWithPos keySerializationStream,
+			final DataOutputView keySerializationDateDataOutputView) throws IOException {
+
 		int length = keySerializationStream.getPosition() - fromPosition;
-		writeVariableIntBytes(length);
+		writeVariableIntBytes(length, keySerializationDateDataOutputView);
 	}
 
-	private void writeVariableIntBytes(int value) throws IOException {
+	private static void writeVariableIntBytes(
+			int value, final DataOutputView keySerializationDateDataOutputView)
```

Comment:
Just a habit of mine so that I don't accidentally change a variable that is not supposed to in the current implementation of the method.
I agree, however, that it is not very consistent among the parameters of each method, e.g. the `value` parameter in your example. I'll remove them.